src/librarySystem/LibraryCLI.java

```diff
@@ -30,17 +31,35 @@ private static void runMainMenu(Scanner userInput, Library library) {
                     break;
 
                 case "2":
+                    int bookId;
                     System.out.println("What book do u want to borrow? Please enter book ID");
-                    int bookId = Integer.parseInt(userInput.nextLine());
+                    while (true) {
+                        bookId = Integer.parseInt(userInput.nextLine());
+                        if (bookId < 0 || bookId > library.getBooks().size()) {
+                            System.out.println("There is no book contains this id, input correct id");
+                            continue;
+                        }
+                        break;
+                    }
+
                     System.out.println("Please input customer's name:");
                     String customerName = userInput.nextLine();
                     Customer customer = new Customer(customerName);
+
+                    LocalDate localDate = null;
                     System.out.println("Please input borrow date (yyyy-mm-dd):");
-                    String borrowDate = userInput.nextLine();
-                    LocalDate date = LocalDate.parse(borrowDate);
-                    library.borrowPrintedProduction(date, bookId, customer);
-                    System.out.println("Book borrowed, it should be returned after a month");
+                    while (localDate == null) {
+                        String borrowDate = userInput.nextLine();
+                        try {
+                            localDate = LocalDate.parse(borrowDate);
+                            library.borrowPrintedProduction(localDate, bookId, customer);
+                            System.out.println("Book borrowed by id " + bookId + ", it should be returned after a month");
```

Comment:
Better print the full book, not only id. Also you have configurable return period, so it worth calculate the exact return date and show it.